### PR TITLE
Use query string URL to create initial map if available

### PR DESCRIPTION
**Pull request by lbebber**
#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/utils/search.js [hunk 1]
```diff
@@ -29,7 +29,29 @@ export const SORT_OPTIONS = [
...
+  if (!url) url = window && window.location && window.location.href
+  name = name.replace(/[\[\]]/g, '\\$&')
+
+  const regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)')
+  const results = regex.exec(url)
+
+  if (!results) return null
+  if (!results[2]) return ''
+
+  return window && decodeURIComponent && window.decodeURIComponent(results[2].replace(/\+/g, ' '))
```
Comment (jgfidelis) [inline react/utils/search.js]: tu não queria fazer `window && window.decodeURIComponent`?

Comment (lbebber) [reply react/utils/search.js]: queria 😅

File: react/utils/search.js [hunk 2]
```diff
@@ -29,7 +29,29 @@ export const SORT_OPTIONS = [
...
+  const regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)')
+  const results = regex.exec(url)
+
+  if (!results) return null
+  if (!results[2]) return ''
+
+  return window && window.decodeURIComponent && window.decodeURIComponent(results[2].replace(/\+/g, ' '))
+}
+
 export function createInitialMap(params) {
```
Comment (luciannojunior) [inline react/utils/search.js]: Couldn't you pass the query here? I mean, where this function is called?

Comment (lbebber) [reply react/utils/search.js]: I don't think there's much to gain by doing that? This is a hotfix anyway, the very need of this `createInitialMap` needs to be reviewed later

Comment (jgfidelis) [reply react/utils/search.js]: Its called in SearchContext.

File: react/utils/search.js [hunk 3]
```diff
@@ -29,7 +29,37 @@ export const SORT_OPTIONS = [
...
+export function initializeMap(params, url) {
+  return getMapFromURL(url) || createInitialMap(params)
+}
+
+export function getMapFromURL(url) {
+  getParameterByName('map', url)
+}
+
 export function createInitialMap(params) {
+  const mapFromUrl = getParameterByName('map')
```
Comment (jgfidelis) [inline react/utils/search.js]: do you still need this after creating and calling `initializeMap`?

Comment (lbebber) [reply react/utils/search.js]: ops not at all

**Pull request merged**